packages/core/src/RenderPipeline/BasicRenderPipeline.ts

```diff
@@ -169,23 +183,28 @@ export class BasicRenderPipeline {
       rhi.clearRenderTarget(camera.engine, pass.clearMode, pass.clearParam);
 
       if (pass.renderOverride) {
-        pass.render(camera, this.queue);
+        pass.render(camera, this._opaqueQueue, this._transparentQueue);
       } else {
-        this.queue.render(camera, pass.replaceMaterial, pass.mask);
+        this._opaqueQueue.render(camera, pass.replaceMaterial, pass.mask);
+        this._transparentQueue.render(camera, pass.replaceMaterial, pass.mask);
       }
 
       rhi.blitRenderTarget(renderTarget);
     }
 
-    pass.postRender(camera, this.queue);
+    pass.postRender(camera, this._opaqueQueue, this._transparentQueue);
   }
 
   /**
    * Push a render element to the render queue.
    * @param element - Render element
    */
   pushPrimitive(element: RenderElement) {
-    this._queue.pushPrimitive(element);
+    if (element.material.renderQueueType >= RenderQueueType.Transparent) {
```

Comment:
Usually the critical point is "(Opaque+Transparent)/2", must not be Transparent.
 For example I want render something transparent  and before the defaut RenderQueueType.Transparent.